### PR TITLE
Extraction of sales data from API added

### PR DESCRIPTION
**Pull request by GabrielNsiah**
The extraction script for just the data from the API is done. For ticket #4. It would be much easier to extract the genre info during the start of the transform phase due to the formatting of the data.

### CODE REVIEW

File: pipeline/extract.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    if response.status_code != 200:
+        logging.warning(
+            "Couldn't retrieve Sales Data. Status Code %s".format(response.status_code))
+        return {}
+    sales_data = response.json()
+    logging.info("Sales Data retrieved.")
+
+    return sales_data
+
+
```
Comment (ebradley12) [inline pipeline/extract.py:29]: Looks good! Can you add genre scraping function with beautiful soup. e.g get_genre_from_url()

Comment (ebradley12) [inline pipeline/extract.py:29]: Add a process data function to include genre information to each sale record

Comment (ebradley12) [reply pipeline/extract.py:32]: We have collectively decided it makes more sense to leave this to the transform script because:
- Adding genre extraction during the extraction phase would blur the lines between extraction and transformation, making the extract function more complex and harder to maintain.
- Not all records from the API are tracks or albums. If we add genre extraction in the extract script, we'd need to clean and filter the data which would then be repeated in transform anyway

File: pipeline/test_extract.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+def test_get_sales_information_failure(mock_get):
+    mock_response = Mock()
+    mock_response.status_code = 500
+    mock_get.return_value = mock_response
+
+    result = get_sales_information()
+
+    assert result == {}
+    mock_get.assert_called_once_with(
+        "https://bandcamp.com/api/salesfeed/1/get_initial")
```
Comment (ebradley12) [inline pipeline/test_extract.py:50]: These tests look good to me. If we have extra time towards the end of the project, we could add more tests for edge cases

File: requirements.txt [hunk 3]
```diff
@@ -0,0 +1,2 @@
+requests 
+beautifulsoup4
```
Comment (ebradley12) [inline requirements.txt]: Add pytest as a requirement

**Review by ebradley12 — APPROVED** [review]

**Comment by ebradley12** [comment]
The pylint workflow is failing - please can you improve the pylint score

**Review by emily-curtis — APPROVED** [review]
Looks really great!!
